pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexCRUDViolation.xml

```diff
@@ -134,6 +134,32 @@ public class Foo {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Proper CRUD,FLS via upsert with database class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public void foo(String newName, String tempID) {
+        if (Contact.sObjectType.getDescribe().isCreateable() && Contact.sObjectType.getDescribe().isUpdateable()) {
+            Database.upsert(new Contact(FirstName = 'First', LastName = 'Last', Phone = '414-414-4414'));
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>No CRUD check for inline upsert with database class</description>
```

Comment:
Could you extend this test case to include the sample code from #2628? Also please add the issue number in the description, this helps us later on to understand, why this test works as it works...

Reply:
The example shown in #2628 is regarding querying records using a dynamic SOQL (string query) which is not being address here (although the comment regarding  DMLs does)

Reply:
Ah, I see. Yes, then, it's a different issue :) Thanks for clarifying.